paymentsheet/src/test/java/com/stripe/android/paymentsheet/CustomerSessionPaymentSheetActivityTest.kt

```diff
@@ -0,0 +1,473 @@
+                        customer = PaymentSheet.CustomerConfiguration.createWithCustomerSession(
+                            id = "cus_1",
+                            clientSecret = "cuss_1",
+                        ),
+                        allowsRemovalOfLastSavedPaymentMethod = allowsRemovalOfLastSavedPaymentMethod,
+                        preferredNetworks = listOf(CardBrand.CartesBancaires, CardBrand.Visa),
+                    ),
+                    statusBarColor = PaymentSheetFixtures.STATUS_BAR_COLOR,
+                )
+            )
+        ).use { scenario ->
+            scenario.onActivity { activity ->
+                composeTestRule.waitUntil(timeoutMillis = 2_000) {
+                    composeTestRule
+                        .onAllNodes(hasTestTag(SAVED_PAYMENT_OPTION_TAB_LAYOUT_TEST_TAG))
+                        .fetchSemanticsNodes()
+                        .isNotEmpty()
+                }
+
+                test(activity)
+
+                countDownLatch.countDown()
+            }
+
+            countDownLatch.await(5, TimeUnit.SECONDS)
+            networkRule.validate()
+        }
+    }
+
+    private fun ComposeTestRule.onEditButton(): SemanticsNodeInteraction {
+        return onNode(
+            hasTestTag(PAYMENT_SHEET_EDIT_BUTTON_TEST_TAG)
+        )
+    }
+
+    private fun ComposeTestRule.onEditScreenRemoveButton(): SemanticsNodeInteraction {
+        return onNode(
+            hasTestTag(PAYMENT_SHEET_EDIT_SCREEN_REMOVE_BUTTON)
+        )
+    }
+
+    private fun SemanticsNodeInteraction.assertHasModifyBadge() {
+        assert(hasAnyDescendant(hasTestTag(TEST_TAG_MODIFY_BADGE)))
+    }
+
+    private fun ComposeTestRule.onModifyBadgeFor(last4: String): SemanticsNodeInteraction {
+        return onNode(
+            hasTestTag(TEST_TAG_MODIFY_BADGE).and(hasAnyAncestor(savedPaymentMethodMatcher(last4)))
+        )
+    }
+
+    private fun ComposeTestRule.onSavedPaymentMethod(last4: String): SemanticsNodeInteraction {
+        return onNode(savedPaymentMethodMatcher(last4))
+    }
+
+    private fun savedPaymentMethodMatcher(last4: String): SemanticsMatcher {
+        return hasTestTag(SAVED_PAYMENT_OPTION_TEST_TAG).and(hasText(last4, substring = true))
+    }
+
+    private fun createCard(last4: String, addCbcNetworks: Boolean = false): PaymentMethod {
+        return PaymentMethodFactory.card(random = true).run {
+            copy(
+                card = card?.copy(
+                    last4 = last4,
+                    networks = PaymentMethod.Card.Networks(
+                        available = setOf("cartes_bancaires", "visa"),
+                        preferred = "cartes_bancaries",
+                    ).takeIf {
+                        addCbcNetworks
+                    }
+                )
+            )
+        }
+    }
+
+    private companion object {
+        @Suppress("LongMethod")
+        fun createElementsSessionResponse(
```

Comment:
JSON string is too long :(, should we add this to the baseline? I worry about making this a habit for future tests though.

Reply:
Another option is to make it a json file, and do the replaces with the replace stuff you introduced the other day.